scripts/hail_batch/hgdp1kg_tobwgs_pca/hgdp_1kg_tob_wgs_pca.py

```diff
@@ -0,0 +1,85 @@
+"""Generates loadings, scores, and eigenvalues for the HGDP,1KG, and tob-wgs dataset"""
+
+import click
+import pandas as pd
+import hail as hl
+from hail.experimental import lgt_to_gt
+
+GNOMAD_HGDP_1KG_MT = (
+    'gs://gcp-public-data--gnomad/release/3.1/mt/genomes/'
+    'gnomad.genomes.v3.1.hgdp_1kg_subset_dense.mt'
+)
+
+TOB_WGS = 'gs://cpg-tob-wgs-main/joint_vcf/v1/raw/genomes.mt'
+
+GNOMAD_LIFTOVER_LOADINGS = 'gs://cpg-reference/gnomad/gnomad_loadings_90k_liftover.ht'
+
+
+@click.command()
+@click.option('--output', help='GCS output path', required=True)
+def query(output):  # pylint: disable=too-many-locals
+    """Query script entry point."""
+
+    hl.init(default_reference='GRCh38')
+
+    # Filter the matrix tables to rows whose keys appear in both datasets
+    mt_hgdp_1kg = hl.read_matrix_table(GNOMAD_HGDP_1KG_MT)
+    mt_tob_wgs = hl.read_matrix_table(TOB_WGS)
+    # make keys the same between datasets
+    mt_tob_wgs = mt_tob_wgs.key_rows_by('locus', 'alleles')
+    join_rows_hgdp1kg_tobwgs = mt_hgdp_1kg.semi_join_rows(mt_tob_wgs.rows())
```

Comment:
Wouldn't an _inner join_ (https://hail.is/docs/0.2/hail.Table.html#hail.Table.join) be the most natural operation if you only want to keep keys that are present in both tables?

Also just double-checking that these are actually the semantics that we want: I guess we can't assume a match with the reference for a locus that's not present for a sample?

Reply:
isn't that function only available for hail tables?

Reply:
Of course, you're absolutely right!